src/worker/hwdetect.rs

```diff
@@ -1,9 +1,70 @@
-use tako::common::resources::{NumOfCpus, ResourceDescriptor};
+use tako::common::resources::{CpuId, NumOfCpus, ResourceDescriptor};
+
+pub fn parse_range(line: &str) -> anyhow::Result<(CpuId, CpuId)> {
+    let line = line.trim();
+    if line.contains('-') {
+        let ids: Vec<&str> = line.split('-').collect();
+        if ids.len() != 2 {
+            anyhow::bail!("Invalid format of range");
+        }
+        Ok((ids[0].parse()?, ids[1].parse()?))
+    } else {
+        let id: CpuId = line.parse()?;
+        Ok((id, id))
+    }
+}
+
+pub fn read_linux_numa() -> anyhow::Result<Vec<Vec<CpuId>>> {
```

Comment:
Maybe we could create a type alias for `Vec<Vec<CpuId>>`? It's used in multiple places and it took me a while to "parse" what it represents.

Reply:
Good point. I have added alias in Tako and I will chage this with the next update of Tako